### PR TITLE
Docs on how to use, what's included, CircleCI config help, Probot, etc.

### PR DESCRIPTION
**Pull request by ruxandrafed**
Here are the main sections I added:
- What's included
- How does this work?
- Why would I want to use CircleCI?
- How do I customize CircleCI for my library?
- How does semantic-release work?
- How does .github/settings.yml work?
- Local development gotchas
- How do I migrate existing libraries?

I included links everywhere for anyone who wants to reference original docs for these tools for more details, included notes on what's needed to have this work locally like in CI, etc. I'll do one more round of checking tomorrow for spelling/grammar issues, etc. and fix if any found.

Please have a look and let me know what you think. Thanks!

P.S. Had to also update some packages to get all jobs to pass in CircleCI 😄

### CODE REVIEW

File: HOW-TO-USE.md [hunk 1]
```diff
@@ -0,0 +1,154 @@
+# The NPM Library Initializer - Step by step guide
```
Comment (ahmadnassri) [inline HOW-TO-USE.md]: this list should live in readme

File: HOW-TO-USE.md [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+- [What's included](#step-1)
+- [How does this work?](#step-2)
+- [Why would I want to use CircleCI?](#step-3)
+- [How do I customize CircleCI for my library?](#step-4)
+- [How does `semantic-release` work?](#step-5)
+- [How does `.github/settings.yml` work?](#step-6)
+- [Local development gotchas](#step-7)
+- [How do I migrate existing libraries?](#step-8)
+
+## <a id="step-1"></a> What's included
```
Comment (ahmadnassri) [inline HOW-TO-USE.md]: this also should be in readme

File: HOW-TO-USE.md [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+- Executing this package means running [index.js](./index.js), which starts the CLI and collects all of your answers.
+- As soon as all answers are collected and minimally processed, they are used to populate placeholders (see [template.js](./lib/template.js)) in the files located inside the `template` folder. 
+- The entire contents of the `template` folder (with the placeholders filled with your info) gets copied in the location where the command was run.
+- Now you can start working on your library!
+
+## <a id="step-3"></a> Why would I want to use CircleCI?
+
+CircleCI is a Continuous Integration tool that allows developers to integrate code into a shared repository several times a day. Each check-in is then verified by an automated build process, allowing the team to detect problems early. Please review our blog post for more details on the benefits of using a CI tool: [How does a CI tool improve Developer Experience?](https://telusdigital.atlassian.net/wiki/spaces/AST/blog/2018/11/01/638877797/How+does+a+CI+tool+improve+Developer+Experience).
```
Comment (ahmadnassri) [inline HOW-TO-USE.md]: I would emphasize that it's the Reference Architecture standard now

File: package.json [hunk 4]
```diff
@@ -41,10 +41,10 @@
     "lint": "npx npm-run-all -p lint:*"
   },
   "dependencies": {
-    "fast-glob": "^2.2.3",
```
Comment (ahmadnassri) [inline package.json]: don't include dependency updates with other changes, it's too disruptive to the release flow

**Comment by ruxandrafed** [comment]
Closing, will reopen with updates for readme (and will keep all the details about CircleCI and semantic release for the `guides` repo.

**Comment by ruxandrafed** [comment]
Reopened as #14 following @ahmadnassri's suggestions.

**Pull request closed without merging**